src/corg/coding-and-design-guidelines/code-review-guidelines.md

```diff
@@ -7,74 +7,235 @@ review:
+- Are the [Coding and Design Guidelines]({{page page='coding-and-design-guidelines'}}) respected?
+- Are there non-regression tests covering the changed code?
+- Does the PR do what it is supposed to do? If it covers more, maybe some code should go to another PR. If it doesn't cover the full scope of the original issue, there might be some missing content.
+- Are there potential security issues?
+- Are there any extra files included in the PR that shouldn't be there?
+- Were new dependencies added to the project? If yes, why were they added?
+- Can I build the branch without any compilation error?
+- Are there any failing tests? If yes, why are they failing?
+- Has the related JIRA issue been updated: description, release notes, upgrade notes?
+- Is there any impact on the documentation to reflect the changes? In this case, do we need an [NXDOC](https://jira.nuxeo.com/projects/NXDOC/) JIRA issue?
 
-Note that this applies to logic or behavior changes that would be interesting to a reader, not cosmetic changes like conflict resolution for imports or slightly different exceptions thrown by a method for instance.
+Obvisouly, we can't expect from a reviewer to address the whole list, neither to always spend a lot of time reviewing a PR. It's up to the reviewers to choose the type of review they are ready to do, for instance:
 
-## When Reviewing Pull Requests
+- "Quick": read the PR globally, check the basic coding guidelines, make sure the non regression tests aren't missing. This shouldn't take more than one or two hours. This can be the case of a reviewer not considering herself as an expert on the topic but can judge if the PR seems globally OK/not OK.
+- "Deep and complete": read the commits in detail, checkout the branch, run the tests, open the code in an IDE, etc. This could take up to several days!
+- "Compromise": a subtle mix of the above mentioned strategies.
 
-Check that tests are present and cover the changed code.
+In any case, a review is always good to have!
 
-In your review, be clear about whether you're requesting a change or you're just mentioning a FYI to the committer.
+### Good Practices
 
-Do not be overly picky: adjust your review level to the level of the coder. Code doesn't have to be perfect, but the goal is still to have it good code quality.
+#### Start on a Positive Note
 
-While it's ok to write short and to-the-point comments, try not to be needlessly abrupt.
+Even if there are a lot of issues with the code that need to be addressed, it's important to thank the person for their contribution and say something nice.
 
-Note that coders and reviewers may be geographically distant, if you cannot do the review side by side, you should just always assume the best intentions: people are trying to help, not to criticize you personally.
+For instance, start with something like "Nice work!" or "Thanks for this great refactoring".
```

Comment:
ok but don't expect that the "praise, correct, praise" attitude is given, sometimes it is culturally difficult :)